src/server/rdb_save.cc

```diff
@@ -1201,28 +1203,58 @@ error_code RdbSaver::Impl::ConsumeChannel(const Cancellation* cll) {
         continue;
 
       DVLOG(2) << "Pulled " << record.id;
-      last_write_time_ns_ = absl::GetCurrentTimeNanos();
-      io_error = sink_->Write(io::Buffer(record.value));
+      auto start = absl::GetCurrentTimeNanos();
+      io_error = WriteRecord(io::Buffer(record.value));
+      if (io_error) {
+        break;  // from the inner TryPop loop.
+      }
 
-      stats.rdb_save_usec += (absl::GetCurrentTimeNanos() - last_write_time_ns_) / 1'000;
+      auto delta_usec = (absl::GetCurrentTimeNanos() - start) / 1'000;
+      stats.rdb_save_usec += delta_usec;
       stats.rdb_save_count++;
       last_write_time_ns_ = -1;
-      if (io_error) {
-        VLOG(1) << "Error writing to sink " << io_error.message();
-        break;
-      }
+
     } while ((channel_.TryPop(record)));
   }  // while (channel_.Pop())
 
   for (auto& ptr : shard_snapshots_) {
     ptr->Join();
   }
-  DVLOG(1) << "Finish ConsumeChannel";
+  VLOG(1) << "ConsumeChannel finished " << io_error;
+
   DCHECK(!channel_.TryPop(record));
 
   return io_error;
 }
 
+error_code RdbSaver::Impl::WriteRecord(io::Bytes src) {
+  // For huge values, we break them up into chunks of upto several MBs to send in a single call,
+  // so we could be more responsive.
+  error_code ec;
+  size_t start_size = src.size();
+  last_write_time_ns_ = absl::GetCurrentTimeNanos();
+  do {
+    io::Bytes part = src.subspan(0, 8_MB);
```

Comment:
In snapshot.cc we have kMinChannelBlobSize.
So each record in channel is at least 32MB size 
Now with this logic you split the writing to sync so for every record we will have at least 4 sink_->Write calls. This does not makes sense to me

Reply:
kMinChannelBlobSize is 32_KB, not 32_MB. They are both needed because `kMinChannelBlobSize` combines multiple small entries, but can not split one huge one. Here we split one huge entry to be able to update `last_write_time_ns_` more frequently.